src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -893,6 +893,38 @@ private PackageSpec GetPackageSpec(IMSBuildProject project, IReadOnlyDictionary<
             restoreMetadata.Sources = GetSources(project, innerNodes, settings);
             restoreMetadata.TargetFrameworks = GetProjectRestoreMetadataFrameworkInfos(targetFrameworkInfos, projectsByTargetFramework);
 
+
+            string skdAnalysisLevelString = project.GetProperty("SdkAnalysisLevel");
+
+            if (skdAnalysisLevelString != null)
```

Comment:
Did you test this?  MSBuild APIs generally treat undefined properties as empty strings, rather than nulls, so I would have thought that `!string.IsNullOrEmpty(sdkAnalysisLevelString)` would be needed. In any case, even if static graph APIs do use null rather than empty string, it might still be worthwhile using `IsNullorEmpty` just to be safe.

Same goes for `isSdk` below.

Reply:
Yes, from my test when the value is not defined, GetProperty returns null as a result if somehow `sdkAnalysisLevelString` is an empty string that would something has gone wrong, and we would throw an exception when we try new NuGetVersion(""). However, we can also just ignore empty strings and nulls as you suggested.

GetProperty : https://github.com/NuGet/NuGet.Client/blob/6c642c2d63717acd4bf92050a42691928020eb89/src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildItemBase.cs#L29